src/python/pants/backend/python/rules/pytest_runner.py

```diff
@@ -49,6 +49,9 @@ class PythonTestFieldSet(TestFieldSet):
     sources: PythonTestsSources
     timeout: PythonTestsTimeout
 
+    def is_conftest(self) -> bool:
+        return not self.address.is_base_target and self.address.filename.endswith("conftest.py")
```

Comment:
I think you want to check equality on the basename, otherwise this will match things like `test_conftest.py`.

Reply:
Also, why the check for `is_base_target`?

Reply:
We check for `is_base_target` because we would otherwise get an exception if we accessed `.filename` and it somehow was not a generated subtarget.

Granted, we should only ever be operating on generated subtargets, but just in case. (For example, our tests do not need to operate on a generated subtarget)